client/service/commands.go

```diff
@@ -1,93 +1,241 @@
 package service
 
 import (
-	"encoding/json"
 	"errors"
 	"fmt"
+	"html"
 	"log"
 	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
 	"time"
 
 	"github.com/databrickslabs/databricks-terraform/client/model"
 )
 
+var (
+	outRE          = regexp.MustCompile(`Out\[[\d\s]+\]:\s`)
+	tagRE          = regexp.MustCompile(`<[^>]*>`)
+	exceptionRE    = regexp.MustCompile(`.*Exception: (.*)`)
+	errorMessageRE = regexp.MustCompile(`ErrorMessage=(.*)\n`)
+)
+
 // CommandsAPI exposes the Context & Commands API
 type CommandsAPI struct {
-	Client *DBApiClient
+	client *DatabricksClient
+}
+
+// CommandMock mocks the execution of command
+type CommandMock func(commandStr string) (string, error)
+
+// CommandExecutorMock simplifies command testing
+type commandExecutorMock struct {
+	mock CommandMock
+}
+
+// Execute mock command with given mock function
+func (c commandExecutorMock) Execute(clusterID, language, commandStr string) (string, error) {
+	return c.mock(commandStr)
 }
 
 // CommandExecutor creates a spark context and executes a command and then closes context
 type CommandExecutor interface {
-	Execute(clusterID, language, commandStr string) (model.Command, error)
+	Execute(clusterID, language, commandStr string) (string, error)
+}
+
+// TrimLeadingWhitespace removes leading whitespace
+func TrimLeadingWhitespace(commandStr string) (newCommand string) {
+	lines := strings.Split(commandStr, "\n")
+	leadingWhitespace := 1<<31 - 1
+	for _, line := range lines {
+		for pos, char := range line {
+			if char == ' ' || char == '\t' {
+				continue
+			}
+			// first non-whitespace character
+			if pos < leadingWhitespace {
+				leadingWhitespace = pos
+			}
+			// is not needed further
+			break
+		}
+	}
+	for i := 0; i < len(lines); i++ {
+		if lines[i] == "" {
+			continue
+		}
+		if len(lines[i]) < leadingWhitespace {
+			newCommand += lines[i] + "\n" // or not..
+		} else {
+			newCommand += lines[i][leadingWhitespace:] + "\n"
+		}
+	}
+	return
 }
 
 // Execute creates a spark context and executes a command and then closes context
-func (a CommandsAPI) Execute(clusterID, language, commandStr string) (model.Command, error) {
-	var resp model.Command
+// Any leading whitespace is trimmed
+func (a CommandsAPI) Execute(clusterID, language, commandStr string) (result string, err error) {
+	cluster, err := a.client.Clusters().Get(clusterID)
```

Comment:
Go doc comment string should start with function name: https://blog.golang.org/godoc

Reply:
it is. first line of comment

Reply:
I meant that the comment should be about what the function is doing.

// Execute will call a command on a databricks cluster given a language. Any leading whitespace on the command string is trimmed.